pkg/controller/daemon/daemoncontroller.go

```diff
@@ -273,6 +289,138 @@ func (dsc *DaemonSetsController) getPodDaemonSets(pod *v1.Pod) []*extensions.Dae
 	return sets
 }
 
+// getDaemonSetsForHistory returns a list of DaemonSets that potentially
+// match a ControllerRevision.
+func (dsc *DaemonSetsController) getDaemonSetsForHistory(history *apps.ControllerRevision) []*extensions.DaemonSet {
+	daemonSets, err := dsc.dsLister.GetHistoryDaemonSets(history)
+	if err != nil || len(daemonSets) == 0 {
+		return nil
+	}
+	if len(daemonSets) > 1 {
+		// ControllerRef will ensure we don't do anything crazy, but more than one
+		// item in this list nevertheless constitutes user error.
+		glog.V(4).Infof("User error! more than one DaemonSets is selecting ControllerRevision %s/%s with labels: %#v",
+			history.Namespace, history.Name, history.Labels)
+	}
+	return daemonSets
+}
+
+// addHistory enqueues the DaemonSet that manages a ControllerRevision when the ControllerRevision is created
+// or when the controller manager is restarted.
+func (dsc *DaemonSetsController) addHistory(obj interface{}) {
+	history := obj.(*apps.ControllerRevision)
+	if history.DeletionTimestamp != nil {
+		// On a restart of the controller manager, it's possible for an object to
+		// show up in a state that is already pending deletion.
+		dsc.deleteHistory(history)
+		return
+	}
+
+	// If it has a ControllerRef, that's all that matters.
+	if controllerRef := controller.GetControllerOf(history); controllerRef != nil {
+		ds := dsc.resolveControllerRef(history.Namespace, controllerRef)
+		if ds == nil {
+			return
+		}
+		glog.V(4).Infof("ControllerRevision %s added.", history.Name)
+		return
```

Comment:
Although the rollout depends on the existence of history, we don't need to requeue on addHistory. 

If the step for creating a new history failed during a rollout, the controller will error out and requeue the DaemonSet (rate limited); if that step succeeded, it'll continue the rollout (creating new pods). If any history is added, the existing DaemonSet won't be affected as long its template isn't changed (trigger a rollout).